### PR TITLE
Draft of DEB packaging

### PR DESCRIPTION
**Pull request by XVilka**
Addressing https://github.com/radareorg/radare2-release/issues/34 and https://github.com/radareorg/radare2-release/issues/35

### CODE REVIEW

File: .github/workflows/packaging.yml [hunk 1]
```diff
@@ -0,0 +1,54 @@
+name: Packaging release assets
+
+on:
+  push:
```
Comment (ret2libc) [inline .github/workflows/packaging.yml:4]: I think we will also need to use `on.release` soon, though I need to check exactly how it works. 

When we choose to start a release we should probably branch with `prerelease-X.Y.Z` and then start all the additional tests, check integration with cutter, r2dec, extras, etc. At that point, we should fix possible issues in master and cherry-pick the commits in the `prerelease-X.Y.Z` branch as well (which will run again all the tests and integration with other repositories). Once all integration tests are fixed, we can create `release-X.Y.Z` branch.

File: .github/workflows/packaging.yml [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+    - name: Making the static build
+      run: sys/static.sh
+      working-directory: radare2
+    - name: Cleaning up the unnecessary files
+      run: git clean -xdf
+      working-directory: radare2
+    - name: Preparing the Debian package
+      run: sys/debian.sh
+      working-directory: radare2
+    - name: Create Release
```
Comment (ret2libc) [inline .github/workflows/packaging.yml:33]: Do you know what happens when we will have multiple jobs that do "Create release"? I'm thinking about jobs for uploading cydia package, android apk, etc. Will it generate multiple releases or re-use the existing one?

Comment (ret2libc) [reply .github/workflows/packaging.yml:33]: Ops sorry, ignore this comment.

File: .github/workflows/packaging.yml [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+
+on:
+  push:
+    branches:
+    - 'pre-release*'
+    - 'prerelease*'
+    - 'release-*'
+
+jobs:
+  build-deb:
```
Comment (ret2libc) [inline .github/workflows/packaging.yml:11]: I think there should be multiple jobs. `build-dep` should be one, then we should have a `create-release` job and a `publish-assets` job that depend on `build-dep`. As done right now, this `build-dep` also create the release and publish the deb package. I think soon we will also have other jobs like `build-android`, `build-cydia`, etc. At that point, each of those jobs will need to be run on a different container.

**Comment by trufae** [comment]
though having debs using my scripts and such was rejected by all you and PPA was the way to go

**Comment by ret2libc** [comment]
> though having debs using my scripts and such was rejected by all you and PPA was the way to go

I still think DEB built by us in this way are not that useful as probably debian maintainers are going to re-create them anyway, but we can't change everything in one shot. I think moving things from bash scripts to gh actions with whatever we currently have is a good first step. This is probably more about a draft for moving release stuff to gh action than strictly about the deb package in itself.

**Comment by XVilka** [comment]
It wasn't rejected - it already builds but with Travis. Also Debian is just
a part of the upcoming release automation.

**Comment by trufae** [comment]
Amazing

**Comment by ret2libc** [comment]
I'm closing this as i'm already working on a more complete solution! I'm going to do a PR when some basic things are done. Thanks!

**Pull request closed without merging**